source/nf_basic.c

```diff
@@ -219,8 +218,13 @@ void NF_SetRootFolder(const char* folder) {
 
 		// Define la carpeta inicial de la FAT
 		sprintf(NF_ROOTFOLDER, "%s", folder);
-		// Intenta inicializar la FAT
-		if (fatInitDefault()) {
+		// Check if DSi SD (sd:/) exists.
+		// If this fails, check if DLDI (fat:/) exists.
+		// If both fails, then fatInitDefault() either failed, or was not called.
+		if (access("sd:/", F_OK) == 0) {
+			// Si es correcto, cambia al ROOT del SD
+			chdir("sd:/");
+		} else if (access("fat:/", F_OK) == 0) {
 			// Si es correcto, cambia al ROOT del FAT
```

Comment:
Do we really need to do this? ``fatInitDefault()`` should probably be in charge of determining the default drive. https://github.com/blocksds/sdk/issues/68

I'm not sure what I would do here, to be honest.

Reply:
Yes, that's correct.

So we can assume that most applications will just want to access the drive where the application is stored. If that's the case, this function should just check ``argv`` to know the location of the NDS file and use that drive as default. If ``argv`` isn't provided, then DSi should default to ``sd:`` and DS to ``fat:``.

``NF_SetRootFolder()`` seems a bit unreliable in general. It stores the provided path in a 32 character array. It should only be used for drive names, then. Also, it doesn't check if the provided path is too long.

How is it supposed to be used? 
``NF_SetRootFolder("NITRO")`` to access assets and then 
``NF_SetRootFolder("/saved/data/location/")`` to save your saved data to FAT?

Reply:
Seems that way.

Fixing the issues you mentioned seems out of scope of this pull request, however. We can try to look at it some other point.

Reply:
Yes, I agree that it's too much for this PR. Let's see if @knightfox75 is ok with the changes.